### PR TITLE
Add Reaper Update script, and add ReaPack to Reaper

### PR DESCRIPTION
**Pull request by Crilum**
closes #1781 (Oh wait, it's already closed :man_facepalming:)

@theofficialgman Isn't there a way to have two version variables? Reaper needs two (or at least could use two), one for Reaper, and one for ReaPack.

### CODE REVIEW

**Comment by theofficialgman** [comment]
yes, see cura for an example
https://github.com/Botspot/pi-apps/blob/master/apps/Cura/install-32
https://github.com/Botspot/pi-apps/blob/master/.github/workflows/updates/Cura.sh

**Comment by Crilum** [comment]
How does this look @theofficialgman?

**Comment by Crilum** [comment]
We might consider having the version variables as Arrays too, so we're not limited to two updatable things per script.

File: etc/categories [hunk 1]
```diff
@@ -128,6 +128,7 @@ Pycharm CE|Editors
 QEMU|Tools
 Quartz|Internet/Browsers
 Raspi2png|Tools
+ReaPack|Multimedia
```
Comment (Botspot) [inline etc/categories:131]: I think this change is unintended and unnecessary.

File: .github/workflows/updates/ReaPack.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (Botspot) [inline .github/workflows/updates/ReaPack.sh:1]: As we're not adding any app named ReaPack, does this update script still belong here?

**Comment by theofficialgman** [comment]
> We might consider having the version variables as Arrays too, so we're not limited to two updatable things per script.

It's not limited to two. webVer are arrays, version I kept as variables. It's automatic as necessary, version version2 version3 version4 .....

File: etc/categories (hunk 1 above)
Comment (Crilum) [reply etc/categories:131]: Oh yeah, I accidentally based this branch off of the ReaPack pr one, so I had to remove ReaPack, but I guess when I did that, I missed the categories file..

File: .github/workflows/updates/ReaPack.sh (hunk 2 above)
Comment (Crilum) [reply .github/workflows/updates/ReaPack.sh:1]: No, I guess I missed that too... lol

File: .github/workflows/updates/Reaper.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+else
+        echo "\$armhf_url ($armhf_url) checks out! We're good to go."
+fi
+
+webVer[0]="${base_ver}${minor_ver}"
+echo "Latest version: ${webVer}"
+
+webVer[1]="$(get_release cfillion/reapack)"
+armhf_url[1]="https://github.com/cfillion/reapack/releases/download/v${webVer}/reaper_reapack-armv7l.so"
+arm64_url[1]="https://github.com/cfillion/reapack/releases/download/v${webVer}/reaper_reapack-aarch64.so"
```
[the remark this answers is not in the document]
Comment (Crilum) [reply .github/workflows/updates/Reaper.sh:38]: Yeah, I'll change that.

**Comment by Crilum** [comment]
I think I've fixed the problems.

I also updated the description to note that this app installs ReaPack, and how to open it.

**Comment by theofficialgman** [comment]
same thing as renoise, you might consider getting the version number directly from the website instead of trying every combination

```
curl -s https://www.reaper.fm/download.php | grep "linux_aarch64.tar.xz"
```
this one is more complicated though than renoise since its the actual webpage and not an index

**Comment by theofficialgman** [comment]
also your current method will break once 7.x releases... because old download links remain and the 6.x link will be found first, and break out of the loop

**Comment by Crilum** [comment]
@theofficialgman

**Comment by theofficialgman** [comment]
not quite what I would suggest. you should only be obtaining the version info once in the update script
and you should be passing that into your download url the same way as you do in your install script.
sec and I'll push a change

**Comment by Crilum** [comment]
:+1:

**Comment by theofficialgman** [comment]
@Botspot do you happen to have a system with a username with spaces in it? I'm not sure if $HOME without being in quotes is safe to use or not

**Comment by Crilum** [comment]
@theofficialgman I don't think you can have spaces in a linux username..

**Comment by theofficialgman** [comment]
> @theofficialgman I don't think you can have spaces in a linux username..

oh no you definitely can have spaces

**Comment by Crilum** [comment]
@theofficialgman 
```
$ sudo adduser "test test"
adduser: To avoid problems, the username should consist only of
letters, digits, underscores, periods, at signs and dashes, and not start with
a dash (as defined by IEEE Std 1003.1-2001). For compatibility with Samba
machine accounts $ is also supported at the end of the username
```

**Comment by theofficialgman** [comment]
```
sudo useradd "test user" --badnames
```
from my memory... there are at least a few distros that when you create the username in the GUI setup they let you add a space, and it must be using this internally

**Comment by theofficialgman** [comment]
and it is dangerous to do it without quotes in this case as confirmed by my test
```
$ echo $HOME
/home/test user
$ mkdir $HOME/hi
mkdir: cannot create directory ‘/home/test’: Permission denied
mkdir: cannot create directory ‘user/hi’: Permission denied
```

**Comment by Crilum** [comment]
This means that all install/uninstall scripts that use `$HOME` should be changed to use `"$HOME"`?

**Comment by theofficialgman** [comment]
> This means that all install/uninstall scripts that use `$HOME` should be changed to use `"$HOME"`?

yeah pretty much...

**Comment by Crilum** [comment]
Well, it wouldn't be *that* hard to made a script the `sed`'s the pi-apps dir and make a PR...

**Comment by theofficialgman** [comment]
> Well, it wouldn't be _that_ hard to made a script the `sed`'s the pi-apps dir and make a PR...

no need for a script... I can just search all in vscode and fix them

**Comment by Crilum** [comment]
Just saying, a script is going to be way faster... There's probably thousands of `$HOME` uses in the `pi-apps` directory...

**Comment by theofficialgman** [comment]
> Just saying, a script is going to be way faster... There's probably thousands of `$HOME` uses in the `pi-apps` directory...

only 300, and most of them are in quotes already which is why you can't script it easily

**Comment by Crilum** [comment]
Yeah, that makes sense.. :+1:

**Comment by Crilum** [comment]
@theofficialgman You'll want to get the `$USER` variables too..

**Comment by theofficialgman** [comment]
ok this becomes a nightmare to fix with $DIRECTORY which is often used in .desktop files which have very cryptic rules for how to use special characters (like spaces https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s07.html )

I can't figure it out and spaces in username isn't posix... lets just hope no users have them

**Comment by Crilum** [comment]
It looks like you can just use double quotes.. 
> Arguments may be quoted in whole. If an argument contains a reserved character the argument must be quoted. ... Reserved characters are space (" ")...

**Comment by theofficialgman** [comment]
> It looks like you can just use double quotes..

thats what the documentation wants you to think... but no
you can try, I already battled with it for 30 minutes and determined its not worth my time

**Comment by Crilum** [comment]
We should put somewhere in the App Creation wiki to use `$HOME` and `$USER` with double quotes..

**Comment by Crilum** [comment]
@Botspot or @theofficialgman Back to the PR subject, I think this is ready to merge..

**Comment by Crilum** [comment]
Thanks  @Botspot!

**Comment by Botspot** [comment]
I wonder how `~/` behaves with a username containing a space character. For `~/`, placing it in quotes actually prevents it from being expanded.

> We should put somewhere in the App Creation wiki to use `$HOME` and `$USER` with double quotes..

I'm not too concerned about keeping our scripts compliant with every possible edge case.
Designing around such limitations makes for complicated and hard-to-read code. For example, there are special scripting techniques for supporting folders that contain newline characters. While it may theoretically improve reliability, it's unlikely to ever be encountered in the real world, so it's just extra programming burden on developers.

For this case however, if several users start complain of broken apps due to a space-separated username, then we may decide to reconsider.

**Pull request merged**